src/main/java/org/example/marketplacebackend/controller/WatchlistController.java

```diff
@@ -1,8 +1,93 @@
+import org.springframework.web.bind.annotation.PostMapping;
+import org.springframework.web.bind.annotation.RequestBody;
+import org.springframework.web.bind.annotation.RequestMapping;
+import org.springframework.web.bind.annotation.RequestPart;
+import org.springframework.web.bind.annotation.RestController;
+import java.security.Principal;
+import java.util.List;
+import java.util.Optional;
+import java.util.UUID;
+
+@RequestMapping("/v1/watchlist")
+@CrossOrigin(origins = { "http://localhost:3000, https://marketplace.johros.dev" }, allowCredentials = "true")
+@RestController
 public class WatchlistController {
+  private final UserService userService;
+  private final WatchListRepository watchListRepository;
+  private final ProductCategoryRepository productCategoryRepository;
+
+  public WatchlistController(UserService userService, WatchListRepository watchListRepository,
+      ProductCategoryRepository productCategoryRepository) {
+    this.userService = userService;
+    this.watchListRepository = watchListRepository;
+    this.productCategoryRepository = productCategoryRepository;
+  }
+
+  @GetMapping("")
+  public ResponseEntity<?> getWatchList(Principal user) {
+    Account authenticatedUser = userService.getAccountOrException(user.getName());
+
+    List<Watchlist> watchList = watchListRepository.findAllBySubscriber(authenticatedUser);
+
+    if (watchList.isEmpty()) {
+      return ResponseEntity.status(HttpStatus.NOT_FOUND).build();
+    }
+
+    List<WatchListResponseDTO> allWatchLists = watchList.stream().map(list -> new WatchListResponseDTO(list.getId(), list.getProductCategory())).toList();
+
+    return ResponseEntity.status(HttpStatus.OK).body(allWatchLists);
+  }
+
+  @PostMapping("")
+  public ResponseEntity<?> postWatchListItem(Principal user, @RequestBody ProductCategory productCategory) {
+
+    if (productCategory == null) {
+      return ResponseEntity.status(HttpStatus.BAD_REQUEST).build();
+    }
+
+    Account authenticatedUser = userService.getAccountOrException(user.getName());
+
+    Watchlist watchList = new Watchlist();
+    watchList.setSubscriber(authenticatedUser);
+    watchList.setProductCategory(productCategory);
+
+    Watchlist returnWatchList = watchListRepository.save(watchList);
+    WatchListResponseDTO watchListResponseDTO = new WatchListResponseDTO(returnWatchList.getId(), returnWatchList.getProductCategory());
+
+    return ResponseEntity.status(HttpStatus.OK).body(watchListResponseDTO);
+  }
+
+  @Transactional
+  @DeleteMapping("/{productCategoryID}")
+  public ResponseEntity<?> deleteWatchListItem(Principal user, @PathVariable UUID productCategoryID) {
+    if (productCategoryID == null) {
+      return ResponseEntity.status(HttpStatus.BAD_REQUEST).build();
+    }
+    Account authenticatedUser = userService.getAccountOrException(user.getName());
+
+    ProductCategory productCategory = productCategoryRepository.findById(productCategoryID).orElseThrow();
```

Comment:
I'd suggest manually checking if the return from the query is successful or not. Throwing an exception means that we give away our control of the response. In this case, it results in an HTTP 403 forbidden.

I'd instead recommend the following approach, as it gives us control over the response:
```java
    ProductCategory productCategory = productCategoryRepository.findById(productCategoryID).orElse(null);
    if (productCategory == null) {
      return ResponseEntity.status(HttpStatus.NOT_FOUND).build();
    }
```